### PR TITLE
fix(select): Clear required & invalid styles #6896

### LINKED ISSUE #6896: [igx-select] Dinamic validation on react form doesn't update UI correctly
## Description  
When setting form validations dynamically, the component `igx-select` doesn't update the UI correctly when switching between 'required' and 'clearValidators'.

 * igniteui-angular version: 9.0.1
 * browser: Chrome Version 80.0.3987.132 (Official Build) (64-bit)

## Steps to reproduce  

1. Open the example at https://stackblitz.com/edit/angular-lbjvac
1.1. In this example, telephone type is only required when a telephone is typed
2. Type anything on the telephone text field and hit "tab" to the next focus. You'll see that the select component became required.
3. Focus back on the telephone text field and clear the value in it, and hit "tab" again to the next focus. This time the select component kept the required UI appearance, even though it's not required anymore.

## Result  
What is the actual result after following the steps to reproduce?
The select component still appears as required, though the form is marked as valid.

## Expected result  
What is the expected result after following the steps to reproduce?
The select component should not appear as required.

## Attachments  
Attach a sample if available, and screenshots, if applicable.  
n/a

### PR DESCRIPTION
**Pull request by IvayloG**
Closes #6896

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/select/select.component.spec.ts [hunk 1]
```diff
@@ -491,18 +493,57 @@ describe('igxSelect', () => {
...
+            expect(inputGroupIsRequiredClass).not.toBeUndefined();
+
+            // Check if the input group's --invalid and --required classes are removed when validator is dynamically cleared
             fix.componentInstance.removeValidators(formGroup);
+            fix.detectChanges();
+            tick();
+
+            inputGroupIsRequiredClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_REQUIRED));
             const selectFormReference = fix.componentInstance.reactiveForm.controls.optionsSelect;
+            // interaction test - expect actual asterisk
```
Comment (ViktorSlavov) [inline projects/igniteui-angular/src/lib/select/select.component.spec.ts:539]: Shouldn't this be "expect MISSING asterisk"?

File: projects/igniteui-angular/src/lib/select/select.component.ts [hunk 2]
```diff
@@ -407,6 +407,13 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
             const error = this.ngControl.control.validator({} as AbstractControl);
             this.inputGroup.isRequired = error && error.required;
             this.cdr.markForCheck();
+
+            // If validator is dynamically cleared,
+            // reset label's required class(asterisk) and IgxInputState #6896
+            } else if (this.inputGroup.isRequired && this.ngControl && !this.ngControl.control.validator) {
```
Comment (ViktorSlavov) [inline projects/igniteui-angular/src/lib/select/select.component.ts:413]: Indentation's off
```suggestion
        } else if (this.inputGroup.isRequired && this.ngControl && !this.ngControl.control.validator) {
```

**Comment by ViktorSlavov** [comment]
This works great w/ the provided scenario in the issue.
Although not a common scenario (and not sure if it's entirely valid), this introduces a new behavior that is not in line w/ the native HTML select:
If the `igx-select` tag has the `required` attribute set in the template AND is specified as a form control, dynamically toggling the validators leads to the asterisk disappearing (so the absent validator takes prevalence over the `required` attribute`).
This is not the case with material and the native select, as can be seen in [this sample](https://stackblitz.com/edit/angular-p4xntp?file=src%2Fapp%2Fselect-hint-error-example.ts)

File: projects/igniteui-angular/src/lib/select/select.component.ts (hunk 2 above)
Comment (IvayloG) [reply projects/igniteui-angular/src/lib/select/select.component.ts:413]: done

File: projects/igniteui-angular/src/lib/select/select.component.spec.ts (hunk 1 above)
Comment (IvayloG) [reply projects/igniteui-angular/src/lib/select/select.component.spec.ts:539]: done

**Comment by IvayloG** [comment]
Now there is an explicit check if somebody has set required HTML attribute as well, for the sake of consistency.

**Pull request merged**